### DISCUSSION TITLE
Adding new(?) Motion Sensor(PIR)

### DISCUSSION THREAD

**Discussion by Jaakko75** [Koenkk/zigbee2mqtt#14758] [Devices]
I'm trying to add a Nedis ZBSM10WT motion sensor. It is made by tuya. Model is TS0202, manufacturerName: '_TZ3000_nss8amz9'. I made the following converter:

`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require("zigbee-herdsman-converters/lib/tuya");

const definition = {
    // Since a lot of Tuya devices use the same modelID, but use different data points
    // it's usually necessary to provide a fingerprint instead of a zigbeeModel
    fingerprint: [
        {
            // The model ID from: Device with modelID 'TS0202' is not supported
            // You may need to add \u0000 at the end of the name in some cases
            modelID: 'TS0202',
            // The manufacturer name from: Device with modelID 'TS0202' is not supported.
            manufacturerName: '_TZ3000_nss8amz9'
        },
    ],
    model: 'ZBSM10WT',
    vendor: 'Nedis',
    description: 'Motion Sensor PIR',
    fromZigbee: [
        fz.ignore_basic_report, // Add this if you are getting no converter for 'genBasic'
        fz.tuya_data_point_dump, // This is a debug converter, it will be described in the next part
    ],
    toZigbee: [
        tz.tuya_data_point_test, // Another debug converter
    ],
    onEvent: tuya.onEventSetTime, // Add this if you are getting no converter for 'commandMcuSyncTime'
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
    },
    exposes: [
        // Here you should put all functionality that your device exposes
    ],
};

module.exports = definition;`

Now I am getting these debug messages: 
Debug 2022-11-01 19:32:34Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Debug 2022-11-01 19:32:34No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":1}'
Debug 2022-11-01 19:32:38Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Debug 2022-11-01 19:32:39No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":0}'

I'm trying to understand what should be done next? I think it should be quite easy, but can someone help bit ?
![image](https://user-images.githubusercontent.com/113261625/199306734-7cd9f7f4-9c3f-4c4f-b0d4-5d2f2322afe1.png)

**Comment by Koenkk** [comment]
Some time after triggering it, do you get another `No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification` (indicating there is no occupancy anymore)?

**Reply by Jaakko75** [reply]
Here is full log from now:

Debug 2022-11-01 21:38:13Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:38:13No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":1}'
Debug 2022-11-01 21:38:19Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:38:19No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":0}'
Debug 2022-11-01 21:38:49Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:38:49No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":1}'
Debug 2022-11-01 21:38:51Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:38:51No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":0}'
Debug 2022-11-01 21:39:19Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:39:19No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":1}'
Debug 2022-11-01 21:39:22Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:39:22No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":0}'
Debug 2022-11-01 21:41:13Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:41:13No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":1}'
Debug 2022-11-01 21:41:15Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:41:15No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":0}'
Debug 2022-11-01 21:41:30Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:41:31No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":1}'
Debug 2022-11-01 21:41:32Saving state to file /config/zigbee2mqtt/state.json
Debug 2022-11-01 21:41:37Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Debug 2022-11-01 21:41:37No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":0}'

**Reply by Jaakko75** [reply]
So it seems that when motion is detected, device sends:

`Debug 2022-11-02 06:54:49Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Debug 2022-11-02 06:54:49No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":1}'`

And when movement is no longer detected it sends:

`Debug 2022-11-02 06:58:35Received Zigbee message from 'PIR', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Debug 2022-11-02 06:58:35No converter available for 'ZBSM10WT' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":0}'`

**Reply by Koenkk** [reply]
Please check with:


```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [{modelID: 'TS0202', manufacturerName: '_TYZB01_dr6sduka'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_ef5xlc9q'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_vwqnz1sn'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_2b8f6cio'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_nss8amz9'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_71kfvvma'},
        {modelID: 'TS0202', manufacturerName: '_TZE200_bq5c8xfe'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_dl7cejts'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_qjqgmqxr'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_mmtwjmaq'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_zwvaj5wy'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_bsvqrxru'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_tv3wxhcz'},
        {modelID: 'TS0202', manufacturerName: '_TYZB01_hqbdru35'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_otvn3lne'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_tiwq83wk'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_ykwcwxmz'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_hgu1dlak'},
        {modelID: 'TS0202', manufacturerName: '_TZ3000_h4wnrtck'},
        {modelID: 'WHD02', manufacturerName: '_TZ3000_hktqahrq'}],
    model: 'TS0202',
    vendor: 'TuYa',
    description: 'Motion sensor',
    whiteLabel: [{vendor: 'Mercator Ikuü', model: 'SMA02P'},
        {vendor: 'TuYa', model: 'TY-ZPR06'},
        {vendor: 'Tesla Smart', model: 'TS0202'}],
    fromZigbee: [fz.ias_occupancy_alarm_1, fz.battery, fz.ignore_basic_report, fz.ias_occupancy_alarm_1_report],
    toZigbee: [],
    exposes: [e.occupancy(), e.battery_low(), e.tamper(), e.battery(), e.battery_voltage()],
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
        try {
            await reporting.batteryPercentageRemaining(endpoint);
        } catch (error) {/* Fails for some https://github.com/Koenkk/zigbee2mqtt/issues/13708*/}
    },
};

module.exports = definition;

```

**Comment by Jaakko75** [comment]
Thank you! Now it works.

**Reply by Koenkk** [reply]
Great, added!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)
